<?xml version="1.0" encoding="UTF-8"?>
<launch>
    
    <arg name="urdf" default="vlp16_bumblebee_vertical" />
    <arg name="tag" default="false"/>

    <include file="$(find safer_stack)/launch/sand_mine.launch"></include>

    <include file="$(find safer_stack)/launch/spawn_husky.launch">
        <arg name="urdf_extras" value="$(find safer_stack)/urdf/$(arg urdf).urdf.xacro" />
    </include>
    
    <include file="$(find safer_stack)/launch/process_camera.launch" />

    <node name="convert_pc_lidar" pkg="safer_stack" type="tf_pointcloud" output="screen">
        <remap from="input" to="/lidar/points2"/>
        <remap from="output" to="/lidar/points2_transformed"/>
    </node>

    <rosparam command="load" file="$(find safer_stack)/config/detection.yaml" />

    

    <!-- <node pkg="safer_stack" name="interface_lidar" type="interface.py" output="screen">
        <remap from="/edge_distance" to="/lidar/edge_distance"/>
        <remap from="/crest" to="/lidar/crest"/>
    </node>

    <node pkg="safer_stack" name="interface_rtabmap" type="interface.py" output="screen">
        <remap from="/edge_distance" to="/rtabmap/edge_distance"/>
        <remap from="/crest" to="/rtabmap/crest"/>
    </node> -->


    <!-- <include file="$(find safer_stack)/launch/rtabmap.launch" /> -->

    <!-- <node name="convert_pc_lidar" pkg="safer_stack" type="tf_pointcloud" output="screen">
        <remap from="input" to="/velodyne_points"/>
        <remap from="output" to="/velodyne/point_cloud_transformed"/>
    </node> -->

    <!-- <node name="convert_pc_camera_filt" pkg="safer_stack" type="tf_pointcloud" output="screen">
        <remap from="input" to="/bumblebee2/statistical_outlier/output"/>
        <remap from="output" to="/bumblebee2/points2_filtered_transformed"/>
    </node> -->

    <!-- <include file="$(find safer_stack)/launch/rtabmap_demo.launch" /> -->

    <!-- <node name="convert_pc_camera" pkg="safer_stack" type="tf_pointcloud" output="screen">
        <remap from="input" to="/bumblebee2/points2"/>
        <remap from="output" to="/bumblebee2/points2_transformed"/>
    </node> -->

    <!-- <node name="ground_detection" pkg="safer_stack" type="ground_detector">
        <remap from="input" to="/bumblebee2/point_cloud_transformed"/>
        <remap from="output" to='/bumblebee2/ground'/>
    </node> -->

    <!-- <rosparam command="load" file="$(find safer_stack)/config/detection.yaml" />

    <node pkg="safer_stack" name="interface" type="interface.py" output="screen"/>
    
    <group if="$(arg tag)">
        <include file="$(find safer_stack)/launch/spawn_apriltag.launch"></include>
        <node pkg="safer_stack" name="distance_tag" type="distance_tag.py" output="screen"/>
    </group>

     -->
     <node name="rviz" pkg="rviz" type="rviz" args="-d $(find safer_stack)/rviz/viz_all.rviz" output="screen"/>

</launch>